File: ch03/swap.zig
const std = @import("std");

fn swap(a: *i32, b: *i32) void {
    // dereference using `.*`
    const temp = a.*;
    a.* = b.*;
    b.* = temp;
}

pub fn main() !void {
    var a: i32 = 45;
    var b: i32 = 32;
    std.debug.print("Before swap, a={}, b={}\n", .{ a, b });
    swap(&a, &b);
    std.debug.print("After swap, a={}, b={}\n", .{ a, b });
}

File: ch03/swap.c
#include<stdio.h>
// swaps two variables using pass-by-reference 
void swap(int* a, int* b){
    // create a temporary value to store a's value 
    // to get int, dereference 
    int temp = *a;
    // store b's value to a
    *a = *b;
    // store a's value to b
    *b = temp;
}

int main(){
    int a = 20;
    int b = 56;
    printf("Before swap, a=%d, b=%d\n", a, b);
    swap(&a, &b);
    printf("After swap, a=%d, b=%d\n", a, b);
    return 0;
}